generate_testdata.sh

```diff
@@ -105,13 +105,10 @@ scaffold_test_project() {
       $kb create webhook --version v1 --kind Lakers --defaulting --programmatic-validation
     fi
   elif [[ $project =~ addon ]]; then
-    header_text 'enabling --pattern flag ...'
-    export KUBEBUILDER_ENABLE_PLUGINS=1
```

Comment:
I am ok with that. However, has few points that I do not agree with here:

**1 - Project layout** 

It represents the layout used to scaffold the project. 
Then, I do not agree with:

```
layout: go.kubebuilder.io/v3,declarative.kubebuilder.io/v1
```

The `declarative.kubebuilder.io/v1` plugin is an extra plugin used to do a scaffold on top of the specific subcommand create api of `golang` plugin, the default project layout. 

Currently, we can decide what APIs will use or not the addon ($ kubebuilder create api --group crew --version v1 --kind Captain --controller=true --resource=true --pattern=addon) and it is scaffolds are not part of the init/base at all. So, I do not agree that we should call this plugin in the init. 

IMO: We ought to
- we can call the plugin in the create api  which means replace `--pattern=addon` by `--plugins=declarative.kubebuilder.io/v1` for the current command `($ kubebuilder create api --group crew --version v1 --kind Captain --controller=true --resource=true --pattern=addon)`
- then, it should be tracked as:

```
plugins:
  declarative.kubebuilder.io/v1: {}
```

Also, I do not think that we should be able/allowed to change/overwrite the default layout. It can be messy and will not allow us to know what was the `base` used to create the project before we define how we will: 
-  described the help
-  define what args will be allowed

I mean, if we scaffold a base `go/v2,helm/v1` will it show the args to create an api valid for the go or helm plugin? Or both? 

Also, the init command should allow we use only plugins which are `base` ones. I mean, we should not able to init a project with a plugin that is not responsible to scaffold the "core/base" files. 

**2 - We need to check and update any documentation over the addon as well.** 
/hold

Reply:
We discussed it a lot today. I will close this comment and add a summary in the review.